marimo/_plugins/ui/_impl/refresh.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2023 Marimo. All rights reserved.
+from __future__ import annotations
+
+from typing import Final, Optional, Union
+
+from marimo._output.rich_help import mddoc
+from marimo._plugins.ui._core.ui_element import UIElement
+
+InternalValue = Union[int, float, str]
+
+
+@mddoc
+class refresh(UIElement[int, int]):
+    """
+    A refresh button that will auto-refresh its descendants for a
+    give interval.
+
+    **Example.**
+    ```python
+    refresh_button = mo.ui.refresh_button(
+        options=["1m", "15m", "30m"],
+        default_value="15m",
+    )
+    ```
+    **Attributes.**
+    - `default_value`: The default value of the refresh interval.
+    - `options`: The options for the refresh interval.
+    - `value`: The time in seconds since the refresh has been activated.
+
+    **Initialization Args.**
+    - `default_value`: The default value of the refresh interval.
+    - `options`: The options for the refresh interval.
+    """
+
+    name: Final[str] = "marimo-refresh"
+
+    def __init__(
+        self,
+        options: list[InternalValue],
```

Comment:
Should `options` be Optional?

So a user can type

`mo.ui.refresh(default_interval="30s")`

instead of

`mo.ui.refresh(options=["30s"], default_interval="30s")`